internal/persistence/sql/uuid_mapping_test.go

```diff
@@ -0,0 +1,58 @@
+package sql_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/gofrs/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ory/keto/internal/driver"
+	"github.com/ory/keto/internal/persistence/sql"
+	"github.com/ory/keto/internal/x/dbx"
+)
+
+func TestUUIDMapping(t *testing.T) {
+	for _, dsn := range dbx.GetDSNs(t, false) {
+		t.Run("dsn="+dsn.Name, func(t *testing.T) {
+			reg := driver.NewTestRegistry(t, dsn)
+			c, err := reg.PopConnection(context.Background())
+			require.NoError(t, err)
+
+			for _, tc := range []struct {
+				desc      string
+				mappings  interface{}
+				shouldErr bool
+			}{{
+				desc:      "empty should fail on constraint",
+				mappings:  &sql.UUIDMapping{},
+				shouldErr: true,
+			}, {
+				desc:      "empty strings should fail on constraint",
+				mappings:  &sql.UUIDMapping{uuid.Nil, ""},
+				shouldErr: true,
+			}, {
+				desc:      "single with string rep should succeed",
+				mappings:  &sql.UUIDMapping{StringRepresentation: "foo"},
+				shouldErr: false,
+			}, {
+				desc: "two with same rep should fail on constraint",
+				mappings: sql.UUIDMappings{
+					&sql.UUIDMapping{StringRepresentation: "bar"},
+					&sql.UUIDMapping{StringRepresentation: "bar"},
+				},
+				shouldErr: true,
+			}} {
+				t.Run("case="+tc.desc, func(t *testing.T) {
+					err = c.Create(tc.mappings)
```

Comment:
```suggestion
					err = sqlcon.HandleError(c.Create(tc.mappings))
```
does some normalization of errors across DBs, but not sure if it also does it for conflict errors. In case it does, you could assert specifically on that error, not just any.

Reply:
done